ssp

```diff
@@ -8424,10 +8472,13 @@ sub _memoize {    ## no critic (RequireArgUnpacking)
 
 sub _memoize_parallel_populate_cache {    ## no critic (RequireArgUnpacking)
                                           # We have to try conditionally loading Data::Dumper here instead of at BEGIN because the CentOS base 'perl' package does not include it, even though it is a core part of Perl.
-    eval { local $SIG{__DIE__}; local $SIG{__WARN__}; require Data::Dumper; };
-    if ($@) {
-        print_info('Perl Data::Dumper: ');
-        print_normal('not available -- SSP will take longer to run -- Use /usr/local/cpanel/3rdparty/bin/perl or "yum install perl-Data-Dumper" if needed');
+    my $dumper = load_module_with_fallbacks(
+        'needed_subs' => [qw{Purity Terse Indent Dump new}],
+        'modules'     => [qw{Data::Dumper}],
+        'fail_safe'   => 1,
```

Comment:
In the interest of reducing the number of necessary options each time it is used, failing gracefully should be the default action (generally everything should try to handle failure gracefully so that SSP can proceed).

In the one place (--profiling option) where we do want to die, maybe add something like 'fail_and_die' => 1?

Reply:
Sounds good. I wasn't sure what we would ultimately want as default for that, so I just picked one :)